### ISSUE TITLE
Packet Provider hangs on device creation though portal reports device is ready

### ISSUE THREAD

**Issue by paemason** [packethost/terraform-provider-packet#238]
This might be the same as #211 

### Terraform Version
terraform -v
Terraform v0.12.23
+ provider.local v1.4.0
+ provider.null v2.1.2
+ provider.packet v2.10.0
+ provider.random v2.3.0
+ provider.template v2.1.2
+ provider.tls v2.1.1

Your version of Terraform is out of date! The latest version
is 0.12.28. You can update by downloading from https://www.terraform.io/downloads.html

### Affected Resource(s)
resource "packet_device"

### Terraform Configuration Files
see https://github.com/packet-labs/google-anthos/blob/master/04-esx-hosts.tf

### Expected Behavior
What should have happened?
Once the server is provisioned, terraform should move to the next step.

### Actual Behavior
What actually happened?
We were stuck here:
packet_device.esxi_hosts[0]: Still creating... [37m10s elapsed]
packet_device.esxi_hosts[0]: Still creating... [37m20s elapsed]
packet_device.esxi_hosts[0]: Still creating... [37m30s elapsed]
packet_device.esxi_hosts[0]: Still creating... [37m40s elapsed]
packet_device.esxi_hosts[0]: Still creating... [37m50s elapsed]
packet_device.esxi_hosts[0]: Still creating... [38m0s elapsed]

**Comment by c0dyhi11** [comment]
This is being used here:
https://github.com/packet-labs/google-anthos/blob/1ac5eadea2aab07b2cadb09fa8741944ee89178b/04-esx-hosts.tf#L1

The API call that is never completing is converting to Mixed/Hybrid. I think this conversion should be moved outside of the device creation and into its own resource. That way if it fails we can re-apply and not have to rebuild the whole server again.

**Comment by c0dyhi11** [comment]
@t0mk Can you take a look at this please? ^^

**Comment by t0mk** [comment]
@c0dyhi11 @paemason Yes, I'll take a look tomorrow.

Is it easy to invoke? Can you prepare a simple template for me to test on? Will just `terraform apply` from the anthos repo work?

**Comment by t0mk** [comment]
@c0dyhi11 @paemason It does indeed sound similar as the previous hybrid/layer3 problems. I might have a clue what's the culprit, I debugged sth similar here: https://github.com/packethost/vault-plugin-secrets-packet/pull/13

**Comment by c0dyhi11** [comment]
@t0mk its intermittent which is the hardest to diagnose. Here is a simple block to test:
```terraform
variable "auth_token" {
}

variable "project_id" {
}

provider "packet" {
  auth_token = var.auth_token
}

resource "packet_reserved_ip_block" "esx_ip_blocks" {
  count      = 3
  project_id = var.project_id
  facility   = "dfw2"
  quantity   = 8
}

resource "packet_device" "esxi_hosts" {
  count            = 3
  hostname         = format("esx-%02d", count.index + 1)
  plan             = "c3.medium.x86"
  facilities       = ["sjc1"]
  operating_system = "vmware_esxi_6_7"
  billing_cycle    = "hourly"
  project_id       = var.project_id
  network_type     = "hybrid"
  tags             = ["vmware", "hypervisor", "anthos"]
  ip_address {
    type            = "public_ipv4"
    cidr            = 29
    reservation_ids = [element(packet_reserved_ip_block.esx_ip_blocks.*.id, count.index)]
  }
  ip_address {
    type = "private_ipv4"
  }
  ip_address {
    type = "public_ipv6"
  }
}
```

**Comment by t0mk** [comment]
@c0dyhi11 thanks. Can I please get enough IP space to project 52000fb2-ee46-4673-93a8-de2c2bdba33b in order to test this?

**Comment by t0mk** [comment]
.. the `terraform apply` fails with
```
Error: The specified project contains insufficient public IPv4 space to complete the request. 
Please e-mail support@packet.com.

```

**Comment by displague** [comment]
If this is related to #211, you might be able to get around it with extended timeouts in the device resource block: https://www.terraform.io/docs/configuration/resources.html#operation-timeouts

The [default for devices is 20m](https://github.com/terraform-providers/terraform-provider-packet/blob/master/packet/resource_packet_device.go#L24-L30).

The only resources that currently support timeouts blocks are devices, spot requests, and volumes.

**Comment by paemason** [comment]
I've tried using extended timeouts and had the same result: the server is provisioned in Packet's portal but stays in layer 3 mode and terraform is stuck waiting for it.

**Comment by t0mk** [comment]
hi, I just verified that this is the same problem as 
https://github.com/packethost/packngo/issues/141

I create the device over TF and then wait. When the API reports it as done, I can see TF still spinning. It's because it's waiting for the network_type to become layer3, so that it can proceed to convert it to `hybrid` (according to the TF template), but the API still reports it (over some endpoints) as `layer2-bonded` which is a pre-init mode, it occurs before the device is ready in layer3 state.

I verified over the API:
```
$ curl -X GET -H "X-Auth-Token: $PACKET_AUTH_TOKEN" https://api.packet.net/devices/cbcbbfeb-71c2-4332-bba6-1129e5e3899c | jq ".network_ports[0].network_type"
"layer2-bonded"
```

According to the API the deivce is in `layer3`. I will keep that^ machine on, so you can check.

It's I think third time we hit this bug.

**Comment by displague** [comment]
The network caching issues reported above should be resolved on the Packet API side.  This should prevent the cases of waiting for an API response that should have already come.

Once the google-anthos module has been updated with #240 (available in 3.0.0 of the Packet Provider), I think we will see a reduction in waits for devices to provision and reach the desired network type. The anthos provisioning steps may have to trigger some scripts only to run after the `packet_device_port_network_type` resource has toggled state and become ready.

https://github.com/packethost/packngo/pull/185 may help in the detection of the current network type, but I am not currently aware of any limitations of the existing GetNetworkType for c2.medium/c3/n2/n3.x-large devices.

**Comment by ocobleseqx** [comment]
**NOTE: This project is Deprecated. The [Equinix provider](https://registry.terraform.io/providers/equinix/equinix/latest/docs) has full support for existing Terraform managed Equinix Metal (Packet) resources once Terraform configuration and state are adapted. The Equinix provider manages resources including Network Edge and Fabric in addition to Metal. [Please review the Packet to Equinix provider migration guide](https://registry.terraform.io/providers/equinix/equinix/latest/docs/guides/migration_guide_packet).**

This issue is outdated and has been inactive for almost 2 years. Network type was decoupled from devices #240. If this issue persist please open a [new issue in Terraform provider equinix repository. ](https://github.com/equinix/terraform-provider-equinix/issues/new?assignees=&labels=&template=bug.md&body=Linked%20issue:%20https://github.com/packethost/terraform-provider-packet/issues/238)
